### ISSUE TITLE
How do I redock a window?

### ISSUE THREAD

**Issue by Moooebie** [nspire-emus/firebird#318]
Hello. I believe this is a very stupid question and not really an "issue" with the program, sorry for posting it here.
But how can I redock a window which is undocked in the UI edit mode? (Just like the keypad shown in screenshot below, which has became a separate window - how can I redock it back to the main window?)
Also I'm having a hard time trying to figure out the keyboard binding (how are the computer keyboard keys mapped to the calculator). I think it would be very helpful to add a section in the wiki about the keyboard shortcuts and key binding...
![image](https://user-images.githubusercontent.com/62904164/235276992-0c521001-49df-4def-8e84-abaf0d3d94e5.png)

**Comment by Vogtinator** [comment]
> But how can I redock a window which is undocked in the UI edit mode? (Just like the keypad shown in screenshot below, which has became a separate window - how can I redock it back to the main window?)

You can drag it over the main window. If you use Wayland, that sometimes just doesn't work, and you need to run firebird with `-platform xcb`...

> Also I'm having a hard time trying to figure out the keyboard binding (how are the computer keyboard keys mapped to the calculator). I think it would be very helpful to add a section in the wiki about the keyboard shortcuts and key binding...

The list is here: https://github.com/nspire-emus/firebird/blob/master/qtkeypadbridge.cpp#L28

Maybe you could start such a wiki page?

**Comment by Moooebie** [comment]
I'm indeed using Wayland, and adding starting the program with `-platform xcb` indeed solved the problem. Thank you!
I'll see if I can figure out the keyboard binding thing and I'll try creating a wiki page...

**Comment by myclevorname** [comment]
Could Firebird override the platform to xcb if it detects that the platform is wayland?

**Comment by Vogtinator** [comment]
Yes, but not sure whether that's a good idea, as wayland works well enough (at least for me)

**Comment by myclevorname** [comment]
I can't dock windows in Firebird using Wayland, and when I move docks to new windows and close them because I can't redock them, I effectively softlock myself until I delete the configuration file.
